plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorOutlinePage.java

```diff
@@ -127,12 +130,14 @@ public void createControl(@NotNull Composite parent) {
         this.treeViewer.setContentProvider(new ILazyTreeContentProvider() {
             @Override
             public void updateElement(@NotNull Object parent, int index) {
-                if (parent == editor.getEditorInput()) {
-                    this.updateChildNode(parent, index, rootNodes.get(index));
-                } else if (parent instanceof OutlineNode node) {
+                if (parent instanceof OutlineNode node) {
                     this.updateChildNode(parent, index, node.getChild(index));
                 } else {
-                    throw new UnsupportedOperationException();
+                    if (index <= rootNodes.size() - 1) {
```

Comment:
This is wrong. Such a condition can never be false while the data model is in sync. Why don't just support the data model contract as it is required by the OutlinePage - so that the data model of the outline is always related with the real state of the owning editor?